README.md

```diff
@@ -35,10 +35,40 @@ import slickeffect.implicits._
 import scala.concurrent.ExecutionContext.Implicits.global
 
 //the instances will be in implicit scope
-scala> Async[slick.dbio.DBIO]
-res0: Async[slick.dbio.package.DBIO] = slickeffect.DBIOAsync@434c179e
+scala> Sync[slick.dbio.DBIO]
+res0: Sync[slick.dbio.package.DBIO] = slickeffect.DBIOSync@434c179e
 ```
 
+### Cats Effect 1.x/2.x -> 3.x migration guide
+
+Cats Effect 3 changes the hierarchy of the type classes that are available, in a way that only lets us implement `Sync`.
```

Comment:
❓ What specifically prevents it? I'm not 100% up to speed on the new hierarchy.

It definitely feels a bit strange.

Reply:
`Async` gives you access to an `ExecutionContext` and `evalOn` (so the ability to replace the EC seen in `def executionContext` - we'd need to slap a reader monad on top of DBIO to get this kind of functionality). Also, `async` and `cancelable` have merged and `async`'s callback itself is now able to return a cancelation token, which would have to be ignored here.

Oh, and that's just the tip of the iceberg - `Async` is also `Temporal`, so it's `Concurrent`. Back to square one - we can't spawn fibers without a database instance (even then, I'm not sure we could - I can give it a try but it's a battle for another day)

Reply:
I discussed this several times on the gitter and starting a new transaction in `start` doesn't seem reasonable - for a DBIO that does `start`, you'd have at least two transactions, which defeats the purpose.

In general I don't think we can get Async anymore. Instead we could provide the ability to lift an effect into a DBIO-like Free-like monad transformer: `DBIOT[F[_], A]`... or provide an explicit LiftIO with a IORuntime, but that requires having an end of world (passing the runtime) every time you lift an IO, which is less than ideal. With a custom effect we could just do it once (although obviously we'll still need to run each IO in the structure individually, it'll be just one "place")

Reply:
I tried this again and it would take implementing basically an entire runloop for something like

`Kleisli[DBIO, (ExecutionContext, Ref[IO, (Finalizers, IsCanceled)], *]`. I gave up halfway because I don't want to be maintaining that if there's a sensible workaround for the lack of Async. And even if we implemented that I'm not sure it would make sense, pass laws or be safe to use.